snapshot/testsuite/testsuite.go

```diff
@@ -18,8 +18,29 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+type SnapshotterSuiteOpt func(*snapshotterSuiteOpts) error
+
+type snapshotterSuiteOpts struct {
+	TestViewReadonly bool
+}
+
+// WithTestViewReadonly ensures a KindView snapshot to be mounted as a read-only filesystem.
```

Comment:
@AkihiroSuda It specifically says "may have the readonly flag set [on the mount]". Roughly, this is saying that the implementation shouldn't rely on `readonly` being set, most will.

For the purposes of existing snapshotters, this is a reasonable assumption. No need to have a combinator here, as most implementers shouldn't be making this choice. Only in the case where we actually have drivers that don't set this flag should we introduce this combinator.

Let's remove the combinator and just test we have readonly flags across the board.